gprofiler/main.py

```diff
@@ -384,6 +375,69 @@ def run_continuous(self) -> None:
                     break
 
 
+def _submit_profile_logged(
+    client: APIClient,
+    start_time: datetime.datetime,
+    end_time: datetime.datetime,
+    profile: str,
+    profile_api_version: Optional[str],
+    spawn_time: float,
+    metrics: "Metrics",
+    gpid: str,
+) -> str:
+    try:
+        response_dict = client.submit_profile(
+            start_time,
+            end_time,
+            profile,
+            profile_api_version,
+            spawn_time,
+            metrics,
+            gpid,
+        )
+    except Timeout:
+        logger.error("Upload of profile to server timed out.")
+    except APIError as e:
+        logger.error(f"Error occurred sending profile to server: {e}")
+    except RequestException:
+        logger.exception("Error occurred sending profile to server")
+    else:
+        logger.info("Successfully uploaded profiling data to the server")
+    return response_dict.get("gpid", "")
```

Comment:
If an exception was raised - `response_dict` is not assigned and this will result in an exception.

I think this code can be placed in `else` clause of the `try`, and at the end of the function, as a fallback, you can use the default value of `""`.